File: isomorphic-strings/isomorphic-strings.cpp
class Solution {
public:
	bool isIsomorphic(string s, string t) {
		unordered_map<char, char> m;
		set<char> st;
		for(int i = 0; i < s.size(); i++){
			if(m.find(s[i]) != m.end()){
				if(m[s[i]] != t[i]){
					return false;
				}
			}
			else{
				if(st.count(t[i]) > 0){
					return false;
				}
				m[s[i]] = t[i];
				st.insert(t[i]);
			}
		}
		return true;
	}
};

File: 205-isomorphic-strings/isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        mp = defaultdict()
        m = defaultdict()
        for a,b in zip(s,t):
            if a in mp:
                if b!=mp[a]:
                    return False
            if b in m:
                if a!=m[b]:
                    return False
            m[b] = a
            mp[a] = b
        return True